lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2193,6 +2195,24 @@ void TypeChecker::checkReferenceOwnershipAttr(VarDecl *var,
     attr->setInvalid();
   }
 
+  ClassDecl *underlyingClass = underlyingType->getClassOrBoundGenericClass();
+  while (underlyingClass != nullptr) {
+    if (auto clangDecl = underlyingClass->getClangDecl()) {
+      if (auto underlyingObjCDecl = cast<clang::ObjCInterfaceDecl>(clangDecl)) {
```

Comment:
Can this cast ever fail, eg if you have a CF type or DispatchQueue or something?

Reply:
Ah, I didn't think of that. Looks like it would fail then. My original intention was to use `dyn_cast` just in case some new language bridge was added (C++? lol). Thanks.

About CF types & Dispatch objects, `isArcWeakrefUnavailable()` is only defined on `clang::ObjCInterfaceDecl` so I assume the attribute it tests for cannot be added to anything else--or at least Clang doesn't support it at the moment.